### ISSUE TITLE
Folder created even if snippet not saved

### ISSUE THREAD

**Issue by jpcirrus** [math2001/BetterSnippetMaker#10]
Thanks for another helpful package.  After using FileManager thought I'd have a look at your other packages 😃

During snippet creation, if the generated snippet is not saved its folder is still created, possibly polluting the parent snippet folder. I believe the full path should only be created when saving the snippet.

**Comment by math2001** [comment]
> Thanks for another helpful package. After using FileManager thought I'd have a look at your other packages

Hey hey, that means I must have done a pretty good job... :slightly_smiling_face: 

Although I'm on vacations right now, it doesn't feel like I have a lot of free time on my hands. Better snippet manager is not the most interesting (in terms of development) nor useful plugin I think. 

It's going to get worse as I'll start university soon, and so I would rather focus fewer projects (like FileManager, reading books, my toread list just grows faster than I can control, playing music, etc).

I'll merge clean PRs, but i don't want to "spread" my energy just working on problems I already know how to approach, and that don't really affect me that much (like I'll happily work on a problem I don't know how to solve :smiley:). It sounds pretty rude when I read this sentence again, but that's the truth, I need to make sure I don't drain myself to at least keep a few things I care about running.

That doesn't mean I'll never fix it, but I'll probably just have one random day when I'll feel like fixing. But there's no guarantee when that will be.

Writing this comment took probably longer than actually fixing the bug, but i'll use this as a reference since it applies to other issues opened on this/other plugins i've written.

I hope you understand :slightly_smiling_face:

**Comment by jpcirrus** [comment]
Yes, no worries. Just finished uni last year and now still busy catching up on what I neglected over the last four years, so I understand, no problem.
